### ISSUE TITLE
PDF tab preview not working.

### ISSUE THREAD

**Issue by Krrost** [James-Yu/LaTeX-Workshop#3760]
Hi,

I have [vscode-server](https://hub.docker.com/r/linuxserver/code-server) in a docker container, hosted on my domain through reverse proxy, with LatexWorkshop and Texlive installed. 

Everything seems to be ok, but I can't see the pdf preview in the tab, there is the pdf viewer, but nothing inside, [here's a screenshot](https://i.stack.imgur.com/14QJ0.png). If I go to get the file, the pdf is generated normally. 

LatexWorkshop log:

```
[Commander] VIEW command invoked with mode: tab.
[Manager] Current workspace folders: ["file://%WS1%","file://%WS2%"]
[Manager] Found root file from active editor: %WS2%/aaa.tex
[Manager] Keep using the same root file: %WS2%/aaa.tex
[Structure] Structure updated with 0 root sections for %WS2%/aaa.tex .
[Viewer][Panel] Internal PDF viewer at https://mydomain.com/proxy/40451/viewer.html?file=pdf..xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx .
[Cacher][Watcher] Watched %WS2%/aaa.pdf .
[Viewer] Open PDF tab for file://%WS2%/aaa.pdf
[Viewer] Handle data type: open
```
The pdf is accessible from the address `https://mydomain.com/proxy/40451/.....`, it correctly shows the [pdf in a browser tab](https://i.stack.imgur.com/cwJnY.png).

So I have a problem with the tab preview, like it can't read the pdf.
Do you happen to know why?

Thanks

**Comment by James-Yu** [comment]
I’ve no idea why. You may need to provide devtools log.

**Comment by James-Yu** [comment]
I’ve no idea why this might happen even with the log. I figure it might be related to your docket network config, yet I don’t think I may further help as I do not have this use case. The strange thing is, the viewer worked fine in browser but not in vscode. Therefore this may also be an issue of vscode, rather than the extension.

**Comment by minghaoL1** [comment]
I came across the same problem (nothing in the pdf tab and the request is also `/proxy/build/pdf.worker.js`), the  PR #3639  did not fix the problem in my case(v9.7.0). I follow the idea in this [issue](https://github.com/coder/code-server/issues/5792#issuecomment-1345499208) to record the proxy port into cookie and rewrite `/proxy/build/pdf.worker.js` to `/proxy/[proxy_port]/build/pdf.worker.js`. It works for me : )

I hosted code server on `example.com/public` and here is my nginx configuration:

<details><summary>configuration</summary>
<p>

```nginx
location ^~/public/ {
  proxy_set_header Host $host;
  proxy_set_header Upgrade $http_upgrade;
  proxy_set_header Connection upgrade;
  proxy_set_header Accept-Encoding gzip;
  proxy_pass http://public/;

  location ~* ^/public/proxy/(?<proxy_port>[^\n\r\/]+)(?<proxy_path>.*)$ {
    default_type text/html;
    if ($proxy_port !~* ^([1-9]|[1-9][0-9]|[1-9][0-9][0-9]|[1-9][0-9][0-9][0-9]|[1-5][0-9][0-9][0-9][0-9]|6[0-5][0-5][0-3][0-5])$) {
      rewrite ^/public/proxy/(.*)$ /public/proxy/$cookie_proxy_port/$1 permanent; # rewrite URI like  /public/proxy/build/pdf.worker.js 
    }
    if ($proxy_path = ''){
      set $proxy_path '/'; #force to use relative path
    }
    add_header Set-Cookie 'proxy_port=$proxy_port; Path=/public';
    proxy_http_version 1.1;
    proxy_pass http://127.0.0.1:$proxy_port$proxy_path;
    proxy_set_header Host $http_host;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Real_IP $remote_addr;
    proxy_set_header User-Agent $http_user_agent;
    proxy_set_header Accept-Encoding "";
    proxy_buffering off;
    #websocket
    set $to_upgrade "";
    set $to_connection "";
    if ($http_upgrade = "websocket"){
      set $to_upgrade "websocket";
      set $to_connection "Upgrade";
    }
    proxy_set_header Upgrade $to_upgrade;
    proxy_set_header Connection $to_connection;
    proxy_read_timeout 86400;
  }
}
``` 

</p>
</details>
